ocaml/xapi/repository_helpers.ml

```diff
@@ -970,7 +975,7 @@ let eval_guidances ~updates_info ~updates ~kind =
       GuidanceStrSet.union acc
         (eval_guidance_for_one_update ~updates_info ~update:u ~kind))
     GuidanceStrSet.empty updates
-  |> Guidance.resort_guidances
+  |> Guidance.resort_guidances ~kind
   |> GuidanceStrSet.elements
   |> List.map Guidance.of_string
```

Comment:
At the moment you are converting from a string set, to a string list, and then doing a list diff later on

We could try to use a set the whole way through, and convert to a json string list only right at the end

(again I would recommend this `module GuidanceSet = Set.Make(struct type t = Guidance.t ;; let compare = Stdlib.compare ) ;;`)

Reply:
Thanks. A new module `GuidanceSet` is added for this.